### PR TITLE
bake: fix print output

### LINKED ISSUE #856: `bake --print` show wrong docker-bake.json
The function to display groups added in #720 does not seem to be working correctly.

The json shown in the following example will not work properly and an error will be displayed.

https://github.com/docker/buildx/blob/bcfd4348290ecb67122a882e0ef42aa537a51ae6/docs/reference/buildx_bake.md?plain=1#L101-L131

```json
{
  "group": {
    "default": [
      "binary"
    ]
  },
  "target": {
    "binary": {
      "context": "git://github.com/docker/cli#master",
      "dockerfile": "Dockerfile",
      "args": {
        "BASE_VARIANT": "alpine",
        "GO_STRIP": "",
        "VERSION": ""
      },
      "target": "binary",
      "platforms": [
        "local"
      ],
      "output": [
        "build"
      ]
    }
  }
}
```

```shell
$ docker buildx bake -f docker-bake.json --print
[+] Building 0.0s (0/0)
docker-bake.json:4
--------------------
   2 |       "group": {
   3 |         "default": [
   4 | >>>       "binary"
   5 |         ]
   6 |       },
--------------------
error: docker-bake.json:4,7-15: Incorrect JSON value type; Either a JSON object or JSON array of objects is required here, to define arguments and child blocks., and 2 other diagnostic(s)
```

Isn't the correct answer as follows?

```json
{
  "group": {
    "default": [
      {
        "targets": [
          "binary"
        ]
      }
    ]
  },
  "target": {
    "binary": {
      "context": "git://github.com/docker/cli#master",
      "dockerfile": "Dockerfile",
      "args": {
        "BASE_VARIANT": "alpine",
        "GO_STRIP": "",
        "VERSION": ""
      },
      "target": "binary",
      "platforms": [
        "local"
      ],
      "output": [
        "build"
      ]
    }
  }
}
```

Also, the contents of groups displayed when groups is specified seems to be wrong.
If we specify the above json and let it `bake --print` again, we will see the json with the default targets as default as shown below.

```json
{
  "group": {
    "default": [
      "default"
    ]
  },
  "target": {
    "binary": {
      "context": "git://github.com/docker/cli#master",
      "dockerfile": "Dockerfile",
      "args": {
        "BASE_VARIANT": "alpine",
        "GO_STRIP": "",
        "VERSION": ""
      },
      "target": "binary",
      "platforms": [
        "local"
      ],
      "output": [
        "build"
      ]
    }
  }
}
```

My local environment is not yet buildx 0.7.0, so I haven't checked it at hand, but I found this behavior on GitHubActions runner.
https://github.com/rocker-org/rocker-versioned2/runs/4277390774?check_suite_focus=true#step:3:5813
(CI to check if docker-bake.json is in the correct format by executing the `bake --print` command)

### PR DESCRIPTION
**Pull request by crazy-max**
fixes #856 

targets groups are not correctly rendered while being printed with `bake --print` so output cannot be reused.

```hcl
// docker-bake.hcl
group "default" {
  targets = ["image"]
}

target "image" {
  inherits = ["image"]
  output = ["type=docker"]
}
```

```json
// docker-bake.json
{
  "group": {
    "default": {
      "targets": [
        "image"
      ]
    }
  },
  "target": {
    "image": {
      "context": ".",
      "dockerfile": "Dockerfile",
      "output": [
        "type=docker"
      ]
    }
  }
}
```

```yaml
# docker-compose.yaml
services:
  addon:
    build:
      context: .
      dockerfile: ./Dockerfile
      args:
        CT_ECR: foo
        CT_TAG: bar
    image: ct-addon:bar
    environment:
      - NODE_ENV=test
      - AWS_ACCESS_KEY_ID=dummy
      - AWS_SECRET_ACCESS_KEY=dummy

  aws:
    build:
      dockerfile: ./aws.Dockerfile
      args:
        CT_ECR: foo
        CT_TAG: bar
    image: ct-fake-aws:bar
```

___

```console
$ docker buildx bake -f docker-compose.yml -f docker-bake.hcl --print
{
  "group": {
    "default": {
      "targets": [
        "image"
      ]
    }
  },
  "target": {
    "image": {
      "context": ".",
      "dockerfile": "Dockerfile",
      "output": [
        "type=docker"
      ]
    }
  }
}
```

```console
$ docker buildx bake -f docker-bake.json --print
{
  "group": {
    "default": {
      "targets": [
        "image"
      ]
    }
  },
  "target": {
    "image": {
      "context": ".",
      "dockerfile": "Dockerfile",
      "output": [
        "type=docker"
      ]
    }
  }
}
```

```console
$ docker buildx bake -f docker-compose.yml -f docker-bake.hcl addon --print
{
  "group": {
    "default": {
      "targets": [
        "addon"
      ]
    }
  },
  "target": {
    "addon": {
      "context": ".",
      "dockerfile": "./Dockerfile",
      "args": {
        "CT_ECR": "foo",
        "CT_TAG": "bar"
      },
      "tags": [
        "ct-addon:bar"
      ]
    }
  }
}
```

```console
$ docker buildx bake -f docker-compose.yml -f docker-bake.hcl addon aws --print
{
  "group": {
    "default": {
      "targets": [
        "addon",
        "aws"
      ]
    }
  },
  "target": {
    "addon": {
      "context": ".",
      "dockerfile": "./Dockerfile",
      "args": {
        "CT_ECR": "foo",
        "CT_TAG": "bar"
      },
      "tags": [
        "ct-addon:bar"
      ]
    },
    "aws": {
      "context": ".",
      "dockerfile": "./aws.Dockerfile",
      "args": {
        "CT_ECR": "foo",
        "CT_TAG": "bar"
      },
      "tags": [
        "ct-fake-aws:bar"
      ]
    }
  }
}
```

```console
$ docker buildx bake -f docker-compose.yml -f docker-bake.hcl addon aws image --print
{
  "group": {
    "default": {
      "targets": [
        "addon",
        "aws",
        "image"
      ]
    }
  },
  "target": {
    "addon": {
      "context": ".",
      "dockerfile": "./Dockerfile",
      "args": {
        "CT_ECR": "foo",
        "CT_TAG": "bar"
      },
      "tags": [
        "ct-addon:bar"
      ]
    },
    "aws": {
      "context": ".",
      "dockerfile": "./aws.Dockerfile",
      "args": {
        "CT_ECR": "foo",
        "CT_TAG": "bar"
      },
      "tags": [
        "ct-fake-aws:bar"
      ]
    },
    "image": {
      "context": ".",
      "dockerfile": "Dockerfile",
      "output": [
        "type=docker"
      ]
    }
  }
}
```

cc @eitsupi

Signed-off-by: CrazyMax <crazy-max@users.noreply.github.com>

### CODE REVIEW

**Review by tonistiigi — COMMENTED** [review]
PTAL Examples in https://github.com/docker/buildx/blob/master/docs/reference/buildx_bake.md

Can we have a test for this?

**Comment by eitsupi** [comment]
Don't you also need a test case for a non "default" group name?

File: bake/bake.go [hunk 1]
```diff
@@ -67,29 +67,53 @@ func ReadLocalFiles(names []string) ([]File, error) {
 	return out, nil
 }
 
-func ReadTargets(ctx context.Context, files []File, targets, overrides []string, defaults map[string]string) (map[string]*Target, []*Group, error) {
+func ReadTargets(ctx context.Context, files []File, inp *Input, targets []string, overrides []string, defaults map[string]string) (map[string]*Target, map[string]*Group, map[string]build.Options, error) {
```
Comment (tonistiigi) [inline bake/bake.go:70]: What's the reason for changing this signature? Can't you just add the group setting and leave the rest of the function as is.

Comment (crazy-max) [reply bake/bake.go]: Thought `TargetsToBuildOpt` had some impact on groups context but it's actually only against targets.

Comment (crazy-max) [reply bake/bake.go]: I will bring back the `[]*Group` signature as before.

Comment (tonistiigi) [reply bake/bake.go]: Generally, I don't mind the map return but as this is going to be picked to release branch probably better to keep the change minimal. We can follow up with the type change in another PR that will not be picked later.

**Review by tonistiigi — COMMENTED** [review]
This doesn't seem to handle 

```
target "default" {}
```

case

**Comment by tonistiigi** [comment]
```
#15 127.5 bake/bake_test.go:453:2: ineffectual assignment to `m` (ineffassign)
#15 127.5 	m, g, err = ReadTargets(ctx, []File{fTargetImage}, []string{"default"}, nil, nil)
#15 127.5 	^
```

**Review by tonistiigi — APPROVED** [review]

**Pull request merged**